pkg/operator/mon/health.go

```diff
@@ -143,6 +143,30 @@ func (c *Cluster) checkHealth() error {
 		return nil
 	}
 
+	// check if there are more than two mons running on the same node, failover one mon in that case
+	nodesUsed := map[string]struct{}{}
+	for name, node := range c.mapping.Node {
```

Comment:
could we use this loop to check for other conditions to failover? For example, if a taint was added to a node to indicate the mon should no longer run there

Reply:
I can add a function to return all now non-valid nodes and check them after the rebalance, is that okay or should that check be before?

Reply:
it is probably better to keep them independent to keep it simple. 
Another thought is that this method is getting rather long. What if we just factored this new code block to a new method?

Reply:
I split up the method into two.